lib/charms/traefik_k8s/v1/ingress.py

```diff
@@ -511,11 +526,21 @@ def url(self) -> Optional[str]:
                 f"Error {e} attempting to read remote app data; "
                 f"probably we are in a relation_departed hook"
             )
-            return None
+            return ""
 
         if not raw:
-            return None
+            return ""
 
         ingress: ProviderIngressData = yaml.safe_load(raw)
         _validate_data({"ingress": ingress}, INGRESS_PROVIDES_APP_SCHEMA)
         return ingress["url"]
+
+    @property
+    def url(self) -> str:
+        """The full ingress URL to reach the current unit.
+
+        May return None if the URL isn't available yet.
+        """
+        data = self._stored.current_url or ""  # type: ignore
+        assert isinstance(data, str)  # for static checker
+        return data
```

Comment:
As in the other PR, shouldn't we change the return type here too?

Reply:
Updated.